python/ray/util/client/dataclient.py

```diff
@@ -52,7 +52,8 @@ def _data_main(self) -> None:
         stub = ray_client_pb2_grpc.RayletDataStreamerStub(self.channel)
         resp_stream = stub.Datapath(
             iter(self.request_queue.get, None),
-            metadata=(("client_id", self._client_id), ))
+            metadata=(("client_id", self._client_id), ),
+            wait_for_ready=True)
```

Comment:
`wait_for_ready` was needed to make the test pass reliably. If we don't do this and the server is started after the client tries to connect, the test will hang.

I'm not sure if we want to do this here though. It would be good to make the client more robust wrt. trying to connect before the server is started, but we probably also want to do so with a timeout.